### PR TITLE
Pathlib adoption for testing.py

### PR DESCRIPTION
**Pull request by cakester**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [x] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [x] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

Phased refactor adoption of loader/segmentation_pair.py as detailed in #872 using pathlib for using pathlib for using pathlib for testing.py

## Linked issues

Step wise remediation of #872

### CODE REVIEW

File: ivadomed/testing.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ def run_inference(test_loader, model, model_params, testing_params, ofolder, cud
                 if last_sample_bool:
                     pred_undo = np.array(pred_undo)
                     if ofolder:
-                        fname_pred = os.path.join(ofolder, fname_ref.split('/')[-1])
+                        fname_pred = str(Path(ofolder, fname_ref.split('/')[-1]))
```
Comment (dyt811) [inline ivadomed/testing.py:280]: Could you replace the `-1` `/` split part with file name?

File: ivadomed/testing.py [hunk 2]
```diff
@@ -47,17 +47,17 @@ def test(model_params, dataset_test, testing_params, path_output, device, cuda_a
...
         model.cuda()
     model.eval()
 
     # CREATE OUTPUT FOLDER
-    path_3Dpred = os.path.join(path_output, 'pred_masks')
-    if not os.path.isdir(path_3Dpred):
-        os.makedirs(path_3Dpred)
+    path_3Dpred = Path(path_output, 'pred_masks')
+    if not path_3Dpred.is_dir():
+        path_3Dpred.mkdir(parents=True)
```
Comment (dyt811) [inline ivadomed/testing.py:60]: This `str` to "path" conversion would require subsequent changes to ensure downstream are not affected. 

Especially the `run_inference(test_loader, model, model_params, testing_params, path_3Dpred, cuda_available, i_monteCarlo, postprocessing)` and `imed_uncertainty.run_uncertainty(ifolder=path_3Dpred)`
for Line 76 and 83 in reference to path_3Dpred

**Review by dyt811 — CHANGES_REQUESTED** [review]
Could you make clarification of the changes?

**Review by dyt811 — APPROVED** [review]
Thanks a lot for taking care of this promptly! 👍
